src/tree/localProject/LocalProjectTreeItem.ts

```diff
@@ -88,15 +89,15 @@ export class LocalProjectTreeItem extends AzExtParentTreeItem implements Disposa
     }
 
     public async getHostJson(): Promise<IParsedHostJson> {
-        const runtime: ProjectRuntime = await this.getRuntime();
+        const version: FuncVersion = await this.getVersion();
         // tslint:disable-next-line: no-any
         const data: any = await fse.readJSON(path.join(this.projectPath, hostFileName));
-        return parseHostJson(data, runtime);
+        return parseHostJson(data, version);
     }
 
-    public async getRuntime(): Promise<ProjectRuntime> {
-        const rawSetting: string | undefined = getWorkspaceSetting(projectRuntimeSetting, this.workspacePath);
-        return nonNullValue(convertStringToRuntime(rawSetting), 'runtime');
+    public async getVersion(): Promise<FuncVersion> {
+        const rawSetting: string | undefined = getWorkspaceSetting(funcVersionSetting, this.workspacePath);
```

Comment:
How is it guaranteed that this is nonNull?

Reply:
Fair question - big picture I'm not too worried about it because I would see this error in telemetry if it was a problem, but I'll look into it after merging this PR (since the behavior didn't change as a part of this PR)